### PR TITLE
convert httpbin to go-based

### PR DESCRIPTION
**Pull request by dhawton**
**Please provide a description of this PR:**

Convert this to a slimmer Go-based httpbin that utilizes port 8080 by default. Also version locked tags to prevent breakages when a new version is released.

### CODE REVIEW

File: samples/httpbin/httpbin-vault.yaml
Comment (craigbox) [inline samples/httpbin/httpbin-vault.yaml:1]: I don't think this file is used anywhere any more.

**Review by craigbox — COMMENTED** [review]
Conceptually pleased with this, haven't tested with any samples though.

**Comment by zirain** [comment]
```
kong/httpbin                                  latest                                            8f751f486162   2 days ago      626MB
mccutchen/go-httpbin                          v2.5.0                                            fbabd38ea186   22 months ago   28.6MB
```

this will save a lot when pulling image

**Review by zirain — APPROVED** [review]

File: samples/httpbin/httpbin-vault.yaml
Comment (dhawton) [reply samples/httpbin/httpbin-vault.yaml:1]: It's in the archives for 1.1 and 1.2, so probably safe to remove.

**Review by bleggett — APPROVED** [review]
LGTM - I see a stray ref for the old-old `httpbin` here?

`pkg/config/analysis/analyzers/testdata/authorizationpolicies.yaml`

**Comment by bleggett** [comment]
Also these still have the `kong` ref:

- samples/extauthz/local-ext-authz.yaml 
- samples/httpbin/README.md

**Review by howardjohn — APPROVED** [review]

**Review by hanxiaop — APPROVED** [review]

**Review by craigbox — APPROVED** [review]

**Pull request merged**